tests/slow/src/test/scala/tests/feature/CompletionCrossLspSuite.scala

```diff
@@ -57,4 +57,52 @@ class CompletionCrossLspSuite
   test("match-213".flaky) {
     matchKeywordTest(V.scala213)
   }
+
+  test("extension") {
+    cleanWorkspace()
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": { "scalaVersion": "${V.scala3}" }
+           |}
+           |/a/src/main/scala/a/B.scala
+           |package b
+           |extension (num: Int)
+           |  def plus(other: Int) = num + other
+           |/a/src/main/scala/a/A.scala
+           |package a
+           |
+           |object A {
+           |  // @@
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/B.scala")
+      _ = assertNoDiagnostics()
+      _ <- assertCompletionEdit(
+        "1.plu@@",
+        """|package a
+           |
+           |import b.plus
+           |
+           |object A {
+           |  1.plus($0)
+           |}
+           |""".stripMargin,
+        filter = _.contains("plus"),
+      )
+      _ <- assertCompletion(
+        "1.plu@@",
```

Comment:
If it's `1.pl@@`, test fails because of the reason I described in the `Tricky part2`.